pettingzoo/butterfly/knights_archers_zombies/knights_archers_zombies.py

```diff
@@ -153,209 +144,108 @@ def action_space(self, agent):
+        ):
+            self.arrow_list.add(Arrow(agent))
 
-    # Zombie Kills the Knight
-    def zombie_knight(self, zombie_list, knight_list, all_sprites, knight_killed, sword_list, sword_killed):
-        for zombie in zombie_list:
-            zombie_knight_list = pygame.sprite.spritecollide(zombie, knight_list, True)
+    # Stab the Sword
+    def move_sword(self):
+        for sword in self.sword_list:
+            sword_active = sword.update()
+            if not sword_active:
+                self.sword_list.remove(sword)
+
+    # Zombie Kills the Knight (also remove the sword)
+    def zombit_hit_knight(self):
+        for zombie in self.zombie_list:
+            zombie_knight_list = pygame.sprite.spritecollide(
+                zombie, self.knight_list, True
+            )
 
             for knight in zombie_knight_list:
                 knight.alive = False
-                knight_list.remove(knight)
-                all_sprites.remove(knight)
-                sword_killed = True
-                knight_killed = True
+                self.knight_list.remove(knight)
                 if knight.agent_name not in self.kill_list:
                     self.kill_list.append(knight.agent_name)
-        return zombie_list, knight_list, all_sprites, knight_killed, sword_list, sword_killed
 
-    # Kill the Sword when Knight dies
-    def kill_sword(self, sword_killed, sword_list, all_sprites):
-        for sword in sword_list:
-            if sword_killed:
-                sword_list.remove(sword)
-                all_sprites.remove(sword)
-                sword_killed = False
-        return sword_killed, sword_list, all_sprites
+                for sword in self.sword_list:
+                    if sword.knight == knight:
+                        self.sword_list.remove(sword)
 
     # Zombie Kills the Archer
-    def zombie_archer(self, zombie_list, archer_list, all_sprites, archer_killed):
-        for zombie in zombie_list:
-            zombie_archer_list = pygame.sprite.spritecollide(zombie, archer_list, True)
+    def zombie_hit_archer(self):
+        for zombie in self.zombie_list:
+            zombie_archer_list = pygame.sprite.spritecollide(
+                zombie, self.archer_list, True
+            )
 
             for archer in zombie_archer_list:
                 archer.alive = False
-                archer_list.remove(archer)
-                all_sprites.remove(archer)
-                archer_killed = True
+                self.archer_list.remove(archer)
                 if archer.agent_name not in self.kill_list:
                     self.kill_list.append(archer.agent_name)
-        return zombie_list, archer_list, all_sprites, archer_killed
 
     # Zombie Kills the Sword
-    def zombie_sword(self, zombie_list, sword_list, all_sprites, score):
-        for sword in sword_list:
-            zombie_sword_list = pygame.sprite.spritecollide(sword, zombie_list, True)
+    def sword_hit(self):
```

Comment:
This doesn't make a lot of sense to me. After watching for awhile I think I might see what this does. The zombie hits the sword, then the sword disappears? Is that what is intended?

Reply:
Yes, this is the intended behaviour, at least this was the behaviour in the previous version. 2 lines down, the zombie that is hit gets deleted too.

Reply:
Yeah, I see this. I don't think it is intended behavior. I don't see it in the documentation. Perhaps @jkterry1 can clear up what the intended behavior is?

Reply:
That's clearly not the intended behavior for the environment